app/models/app_remote_kill.rb

```diff
@@ -0,0 +1,65 @@
+class AppRemoteKill < ActiveRecord::Base
+  belongs_to :app
+  belongs_to :split
+
+  attribute :first_bad_version, :app_version
+  attribute :fixed_version, :app_version
+
+  validates :app, :split, :reason, :override_to, :first_bad_version, presence: true
+  validates :reason, uniqueness: { scope: %i(app split) }
+
+  validate :override_to_must_exist
+  validate :fixed_version_must_be_greater_than_first_bad_version
+  validate :must_not_overlap_existing
+
+  scope :affecting, ->(app_build) do
+    where(
+      arel_table[:app_id].eq(app_build.app_id)
+      .and(arel_table[:first_bad_version].lteq(app_build.version))
+      .and(arel_fixed_version_is_null_or_greater_than(app_build.version))
+    )
+  end
+
+  scope :overlapping, ->(other) do
+    where(app_id: other.app_id, split_id: other.split_id)
+      .where.not(id: other.id)
+      .where(
+        if other.fixed_version.nil?
+          Arel::Nodes::True.new
+        else
+          arel_table[:first_bad_version].lt(other.fixed_version)
+        end
+        .and(arel_fixed_version_is_null_or_greater_than(other.first_bad_version))
+      )
+  end
+
+  def self.arel_fixed_version_is_null_or_greater_than(version)
+    Arel::Nodes::Grouping.new(
+      Arel::Nodes::Or.new(
```

Comment:
TIL that Or nodes are just that - OR'd together nodes in the AST - they don't confer parens around the OR, which is very unsafe, but hey, that's how SQL works. So you'll notice that I wrapped Grouping nodes (Arel for "parens") around all the Or nodes I'd been building elsewhere.

Reply:
Feels like Arel is in need of a more concise DSL that also doesn't hide all of these features like AR's query builder.

Reply:
(I know there have been a few attempts over the years, but even some aliasing to more terse method names might help.)

Reply:
I just switched back to infix or syntax actually, which partially solves. I still need to use gnarly `Arel::Nodes::Grouping.new` but can just use `.or(other)` on a variety of arel node classes which reads better.